### ISSUE TITLE
Latest Version (6.5.0) of AppLovinQualityServiceSetup.rb Fails with AppLovin 5.4.3

### ISSUE THREAD

**Issue by MBreitkreutz** [AppLovin/AppLovin-MAX-Unity-Plugin#170]
It appears that AppLovinQualityServicePod2 was updated to v6.5.0 on June 12. After this update, iOS builds get the following error when running AppLovinQualityServiceSetup.rb:

```
Error [AppLovin MAX] Failed to set up AppLovin Quality Service: Failed to run command: 'ruby AppLovinQualityServiceSetup.rb'
stdout: AppLovin Quality Service Xcode Setup Script
Copyright (c) 2022 AppLovin. All Rights Reserved.
Script version: 6.5.0
Executing script version: 6.5.0
AppLovinQualityService Pod has the latest version, no update required
Extracting Pod AppLovinQualityServicePod2-6.5.0.zip...


stderr: Swift/ErrorType.swift:200: Fatal error: Error raised at top level: AppLovinQualityServiceInstaller.ProjectFileError.internalInconsistency([AppLovinQualityServiceInstaller.ReferenceError.deadReference(type: "PBXTargetDependency", id: AppLovinQualityServiceInstaller.Guid(value: "5623C58217FDCB0900090B9E"), keyPath: "targetProxy", ref: AppLovinQualityServiceInstaller.Guid(value: "5623C58117FDCB0900090B9E")), AppLovinQualityServiceInstaller.ReferenceError.deadReference(type: "PBXTargetDependency", id: AppLovinQualityServiceInstaller.Guid(value: "9D25ABA3213FB47800354C27"), keyPath: "targetProxy", ref: AppLovinQualityServiceInstaller.Guid(value: "9D25ABA2213FB47800354C27")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "1CF74509A970271FB49B1D40")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "217A446FA1A0368924ACBABC")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "24CA45F4A6DFC15E54BB30AD")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "3DEF4D6AB407268A3DF6B397")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "3E8C4637B09EEEA915B06740")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "401C4DCDBA397CC65CAAF803")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "44D04AC8B0013A68028DE555")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "4A8644E892CA9B9D64AA4D04")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXFileReference", id: AppLovinQualityServiceInstaller.Guid(value: "5623C57C17FDCB0900090B9E")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "5F644E0E8BBF7043AA5D8245")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "68A64F4D8681B16795D6AD44")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "76394B44940227D16EC87FD4")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "786845D7A2097EB9787BE9C6")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "8328409790972CADD1F6E7DC")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "891C42DB9CC50A7785E772E6")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "92D94EA2BAD142C0ED7E24A4")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "9D25ABA1213FB47800354C27")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "9DC67E8521CBBEBB005F9FA1")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "9DC67E8621CBBEC7005F9FA1")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "9DC67E8721CBBEC7005F9FA1")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "9DC67E8821CBBED5005F9FA1")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "9DC67E8921CBBEDF005F9FA1")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "B3424A73882F930D07CC1923")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "C77943D89D516040679FB8EF")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "CF5F439CAE2299675909A674")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "DEF844C496EEEC86FB675D9F")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "E2934AD687815F3A992364F2")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "E383441BBB947CF16CC70F75")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "ED024B8593B8F2F631ACCEBE")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "FA2F41EE9DAE13ADD270FF57")), AppLovinQualityServiceInstaller.ReferenceError.orphanObject(type: "PBXBuildFile", id: AppLovinQualityServiceInstaller.Guid(value: "FDEE4110814DA55171B1AFB1"))])

Failed to apply the AppLovinQualityService Pod
AppLovinQualityService setup FAILED


Exit code: 1
```

I manually ran the previous version (v6.4.3) of the script and verified that it still works properly and iOS builds using it do not report this error. I updated to the latest version of the AppLovin Unity plugin (v5.4.3, Android v11.4.3, iOS v11.4.2) to verify the issue persists (was previously on Unity plugin v5.4.2).

I will have to continue to use v6.4.3 of the ruby script until there is a resolution for this.

**Comment by orilentz** [comment]
Thanks for reporting. We've released a hotfix for this issue - 6.5.1. If you recompile, you should see the latest version pulled in the logs or Mediation Debugger

**Comment by jezun3twork** [comment]
Hi @orilentz builds are failing on my side too using AppLovin 5.4.3. Not seeing this 6.5.1 version of AppLovinQualityServicePod2 (what you mean by recompile?)

```
AppLovinQualityService Pod was updated within the last 24 hours, no update required
Extracting Pod AppLovinQualityServicePod2-6.5.0.zip...

stderr: Swift/ErrorType.swift:200: Fatal error: Error raised at top level: AppLovinQualityServiceInstaller.ProjectFileError.internalInconsistency
```

**Comment by MBreitkreutz** [comment]
> Thanks for reporting. We've released a hotfix for this issue - 6.5.1. If you recompile, you should see the latest version pulled in the logs or Mediation Debugger

Thanks, the hotfix works as expected. Version 6.5.1 is used and no errors are reported.

**Comment by orilentz** [comment]
@jezun3twork if you still get 6.5.0, that might mean the export isn't fully clean. Make sure you delete the exported project from the filesystem before cleaning and exporting again.
